### PR TITLE
[ebpf] Fix connection expiration race condition

### PR DESCRIPTION
**Pull request by p-lambert**
### What does this PR do?

This PR fixes a connection expiration data race that was causing network metrics to be over-reported in some cases.

Basically the data race arises in the following scenario:
1. Connection check is executed (`Tracer.GetActiveConnections()`)
2. One of the connections (let's call it *c* ) hasn't been updated in the eBPF map for for longer than ` TCPConnTimeout` and it's "marked" as expired (`Tracer.getConnections`);
3. At the same time the connection *c* is closed and `kprobe/tcp_close` is executed;
4. The closed connection is written to the perf ring buffer. Once user space reads it, it calls `state.StoreClosedConnection` (from `Tracer.initPerfPooling`);
5. Now the function call initiated in **2.** calls `Tracer.removeEntries` with connection *c* as an argument;
6. The state associated to *c* is removed;
7. Client calls again `Tracer.GetActiveConnections()`; Since there is no state associated to *c* but just the `closedConnection` entry, the user gets back `Monotonic{Sent,Recv}Bytes` as if it was `Last{Sent,Recv}Bytes`

#### Note about the test

The testing code is unfortunately too tied to the private API of the tracer, so I'm not sure if it's totally worth it. Another way to test this would be setting a small number to `TCPConnTimeout`, opening a bunch of connections and then closing them and calling `GetActiveConnections` after `TCPConnTimeout` seconds. That would potentially trigger the race condition without relying on the private API, but the test could become too flaky so 🤷‍♂ 

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

**Review by kevinconaway — DISMISSED** [review]
LGTM but I think you need to run `go fmt`

File: pkg/ebpf/tracer.go [hunk 1]
```diff
@@ -438,8 +438,13 @@ func (t *Tracer) removeEntries(mp, tcpMp *bpflib.Map, entries []*ConnTuple) {
 	for i := range entries {
 		err := t.m.DeleteElement(mp, unsafe.Pointer(entries[i]))
```
Comment (leeavital) [inline pkg/ebpf/tracer.go]: would DeleteMap return an error under any other circumstance?

**Review by leeavital — DISMISSED** [review]

File: pkg/ebpf/tracer.go (hunk 1 above)
Comment (kevinconaway) [reply pkg/ebpf/tracer.go]: [Last time](https://github.com/iovisor/gobpf/pull/186), I looked at this, it [returns an error[(https://github.com/iovisor/gobpf/blob/master/elf/table.go#L130-L137) if the syscall fails or if the element is not found

Comment (p-lambert) [reply pkg/ebpf/tracer.go]: In either case I think it's safe to keep the code as it is.
If it's a system call failure, which isn't exactly what we're expecting, we'll re-attempt to delete the entry in the next `GetActiveConnections` call.

**Review by sunhay — DISMISSED** [review]
This looks great - if we continue to see some issues, it'll likely be worth adding a lock around access to the state object between the perf buffer + the map iterator.

**Review by sunhay — APPROVED** [review]

**Pull request merged**